gaphor/codegen/profile_coder.py

```diff
@@ -0,0 +1,85 @@
+"""Parse a SysML Gaphor Model and generate a SysML data model."""
+
+from typing import List, Optional, Set
+
+from gaphor import UML
+from gaphor.application import Session
+from gaphor.storage import storage
+
+
+def type_converter(association) -> Optional[str]:
+    if association.typeValue is None:
+        return None
+        # raise ValueError(
+        #     f"ERROR! type is not specified for property {association.name}"
+        # )
+    if association.typeValue.lower() == "boolean":
+        return "int"
+    elif association.typeValue.lower() in ("integer", "unlimitednatural"):
+        return "int"
+    elif association.typeValue.lower() == "string":
+        return "str"
+    else:
+        return str(association.typeValue)
+
+
+def write_attributes(cls: UML.Class, filename) -> None:
+    if not cls.attribute or not cls.attribute[0].name:
+        filename.write("    pass\n\n")
+    else:
+        for a in cls.attribute["not it.association"]:  # type: ignore
+            type_value = type_converter(a)
+            filename.write(f"    {a.name}: attribute[{type_value}]\n")
+        for a in cls.attribute["it.association"]:  # type: ignore
+            filename.write(f"    {a.name}: association\n")
+        for o in cls.ownedOperation:
+            filename.write(f"    {o}: operation\n")
+
+
+def generate(filename, outfile=None, overridesfile=None):
+    services = [
+        "element_dispatcher",
+        "event_manager",
+        "component_registry",
+        "element_factory",
+        "modeling_language",
+    ]
+    session = Session(services=services)
+    element_factory = session.get_service("element_factory")
+    modeling_language = session.get_service("modeling_language")
```

Comment:
I made a small fix on master, so we can just do:

```
element_factory = ElementFactory()
modeling_language = UMLModelingLanguage()
```

Reply:
Thanks for updating that! I'll merge master back in to this branch and start to make use of that.